src/main/scala/com/databricks/spark/xml/DefaultSource.scala

```diff
@@ -67,11 +67,23 @@ class DefaultSource
       (options.charset, options.rowTag)
     }
 
-    XmlRelation(
-      () => XmlFile.withCharset(sqlContext.sparkContext, path, charset, rowTag),
-      Some(path),
-      parameters,
-      schema)(sqlContext)
+    val sparkTimezone = sqlContext.sparkContext.getConf.getOption(
+      "spark.sql.session.timeZone"
+    )
+
+    if (sparkTimezone.isDefined) {
```

Comment:
Can you just vary `parameters` based on `sparkTimezone` rather than repeat the whole block?